source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp

```diff
@@ -172,8 +179,8 @@ bool MaterialXNode::setInternalValue(const MPlug &plug, const MDataHandle &dataH
 				}
 				else if (type == MaterialX::TypedValue<MaterialX::Vector4>::TYPE)
 				{
-					double4& value = dataHandle.asDouble4();
-					valueElement->setValue(MaterialX::Vector4(static_cast<float>(value[0]), static_cast<float>(value[1]), static_cast<float>(value[2]), static_cast<float>(value[3])));
+					//double4& value = dataHandle.asDouble4();
```

Comment:
Even after rebuilding, this does not compile for me as there is no asDouble4() in my version of Maya. We need to be mindful of the version we use.

Reply:
It's present in Maya 2019. What's our minimum Maya version requirement?

Reply:
If this could work in 2018 that would be good. Guess we'll need to #ifdef this.